pubsub/gossipsub/README.md

```diff
@@ -54,11 +54,11 @@ It implements pubsub in the most basic manner, with two defining aspects:
 
 ### Ambient Peer Discovery
 
-With ambient peer discovery, the function is pushed outside the scope
-of the protocol. Instead, it relies on ambient connection events to
-perform peer discovery via protocol identification. Whenever a new
+In ambient peer discovery, discovering other pubsub peers is a passive action. 
```

Comment:
please don't change the original sentence; you can add a clarifying sentence on top.

Reply:
I think the original sentence is lacking. It doesn't specify what function is pushed outside the scope of the protocol. If the function you are referring to is peer discovery then you should state it. Otherwise, the sentence doesn't make much sense.

Reply:
It makes perfect sense to me, i am not going to change it... but I am fine with adding a clarification.

Reply:
Of course it makes perfect to you, you wrote the specification. However, when writing specifications, you aren't writing them for yourself but for others. If you add a clarification for what "the function" pertains to, then you can keep the sentence with the added clarification.

Reply:
the function is the topic of the section; ambient peer discovery.

Reply:
BTW – we’re missing an intro here. That’s my fault! @vyzo, @mikerah is implementing gossipsub-js from scratch, and she would like to rely on the spec as much as possible, but is having to fall back to the implementation very frequently for clarifications. So I think her fresh view will bring maturity and help us make this important spec more accessible to a wider audience.

Reply:
Any updates on this PR?

Reply:
I think it's quite circular. Upon a first reading of it, it's quite confusing. I only started to get it after multiple reads and opening an issue in this repository. I'm pretty sure there are others that might feel the same way.

Reply:
If you feel that way, you are very welcome to add a clarifying sentence :)

Reply:
I updated your sentence to be more clear.

Reply:
you are still changing the original sentence, which I very much like.  Please don't.

Reply:
I think the new sentence is imprecise in the sense that "discovery of pubsub peers" is too coarse grained. This can be decomposed in "discovering a peer" and "identifying it's a compatible pubsub peer". I do see how the original sentence is confusing. May I suggest this change?

```suggestion
With ambient peer discovery, the mechanism for discovering peers is provided for by the environment. In practice, this can be embodied by DHT walks, rendezvous points, etc. This protocol relies on the ambient connection events produced by such mechanisms. Whenever a new peer is connected, the protocol checks to see if the peer implements floodsub and/or gossipsub, and if so, it sends it a hello packet that announces the topics that it is currently subscribing to. 
```

Reply:
And can we please leave my sentence unmolested? I am issuing a veto here :)

Reply:
Molestation reverted 😛 I added a few clarifying sentences in https://github.com/libp2p/specs/pull/127/commits/ecf4623b13bba68e108185b7c78ae787094ad3d0. LGTY?